src/main/java/de/tum/in/www1/artemis/service/SubmissionPolicyService.java

```diff
@@ -307,18 +307,25 @@ else if (submissions > allowedSubmissions) {
      * Calculates and returns the number of submissions for one participation. This amount represents
      * the amount of unique manual submissions with at least one result.
      *
-     * @param participation for which the number of submissions should be determined
+     * @param participation              for which the number of submissions should be determined
+     * @param withSubmissionCompensation flag that indicates that the newest result is not yet saved and should be accounted for
      * @return the number of submissions of this participation
      */
-    private int getParticipationSubmissionCount(Participation participation) {
-        final Long participationId = participation.getId();
+    public int getParticipationSubmissionCount(Participation participation, boolean withSubmissionCompensation) {
+        long participationId = participation.getId();
+
+        // The newest result is not yet stored in the database. Therefore, we have to add one additional submission if the latest submission does not have a result yes, since it
```

Comment:
```suggestion
        // The newest result is not yet stored in the database. Therefore, we have to add one additional submission if the latest submission does not have a result yet, since it
```